include/WaveFunction.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef GQCG_WAVEFUNCTION_HPP
+#define GQCG_WAVEFUNCTION_HPP
+
+
+#include "FockSpace/BaseFockSpace.hpp"
+
+#include "common.hpp"
+
+
+
+namespace GQCG {
+
+
+class WaveFunction {
+private:
+    BaseFockSpace* base_fock_space;
+    Eigen::VectorXd coefficients;  // Expansion coefficients of a wave function in the Fock space
+
+public:
+    // CONSTRUCTORS
+    WaveFunction(BaseFockSpace& base_fock_space, const Eigen::VectorXd& coefficients) : base_fock_space(&base_fock_space), coefficients(coefficients){}
```

Comment:
const reference

Reply:
will have to be done in a 'const correct feature' currently this would have to make the BaseFockSpace* a const BaseFockSpace* and the getters would be required to return a const BaseFockSpace& and there for any functions called upon this FockSpace would have to be const functions, etc.